lms/djangoapps/api_manager/courses/tests.py

```diff
@@ -1330,22 +1332,69 @@ def test_courses_users_list_get_attributes(self):
-        self.assertIn('url', response.data['enrollments'][0]['organizations'][0])
-        self.assertIn('id', response.data['enrollments'][0]['organizations'][0])
-        self.assertIn('name', response.data['enrollments'][0]['organizations'][0])
-        self.assertIn('created', response.data['enrollments'][0]['organizations'][0])
-        self.assertIn('display_name', response.data['enrollments'][0]['organizations'][0])
-        self.assertIn('logo_url', response.data['enrollments'][0]['organizations'][0])
-        roles = response.data['enrollments'][0]['roles']
+        self.assertIn('id', response.data['results'][0])
+        self.assertIn('email', response.data['results'][0])
+        self.assertIn('username', response.data['results'][0])
+        self.assertIn('last_login', response.data['results'][0])
+        self.assertIn('full_name', response.data['results'][0])
+        self.assertIn('is_active', response.data['results'][0])
+        self.assertIsNotNone(response.data['results'][0]['organizations'])
+        self.assertIn('url', response.data['results'][0]['organizations'][0])
+        self.assertIn('id', response.data['results'][0]['organizations'][0])
+        self.assertIn('name', response.data['results'][0]['organizations'][0])
+        self.assertIn('created', response.data['results'][0]['organizations'][0])
+        self.assertIn('display_name', response.data['results'][0]['organizations'][0])
+        self.assertIn('logo_url', response.data['results'][0]['organizations'][0])
+        roles = response.data['results'][0]['roles']
         self.assertIsNotNone(roles)
         self.assertEqual(['instructor', 'observer'], roles)
+        self.assertIn('grades', response.data['results'][0])
+        self.assertEqual(
+            response.data['results'][0]['grades'], {'grade': user_grade, 'proforma_grade': user_proforma_grade}
+        )
+
+    def test_courses_users_list_with_fields(self):
+        """ Tests when fields param is given it should return only those fields """
+        course = CourseFactory.create()
+        users = UserFactory.create_batch(3)
+        for user in users:
+            CourseEnrollmentFactory.create(user=user, course_id=course.id)
+            user_grade, user_proforma_grade = 0.9, 0.91
+            StudentGradebook.objects.create(user=user, course_id=course.id,
+                                            grade=user_grade, proforma_grade=user_proforma_grade)
+            allow_access(course, user, 'instructor')
+
+        test_uri = self.base_courses_uri + '/' + unicode(course.id) + '/users?fields=first_name,last_name'
+        response = self.do_get(test_uri)
+        self.assertEqual(response.status_code, 200)
+        self.assertEqual(len(response.data['results']), 3)
+        self.assertIn('first_name', response.data['results'][0])
+        self.assertIn('last_name', response.data['results'][0])
+        self.assertNotIn('id', response.data['results'][0])
```

Comment:
i think instead of testing all attributes separately we can do something like ``self.assertItemsEqual(['first_name', 'last_name'], response.data['results'][0].keys())``